### DISCUSSION TITLE
Continuous cutting boolean

### DISCUSSION THREAD

**Discussion by hrgdavor** [elalish/manifold#1012] [Ideas]
There is a use-case for booleans that I do not think can be implemented by interfacing with manifold, but would need something internal to support optimisation.

One such case is simulating cnc tools path and how it affects a slab. It would enatial a starting geometry, that is kept in memory with associated data possible, and then produce outputs by code giving a boolean op (some shape to cut) and returning current mesh.

other such use-case has been presented when ppl got excited by perf boost in babylonjs, that they got from switching from csg.js to manifold.

https://forum.babylonjs.com/t/introducing-csg2/54274/26
https://playground.babylonjs.com/?snapshot=refs/pull/15713/merge#IW43EB#22

![a17bb48dad3bb602ae0734e1c9ca3a16299d1ea2](https://github.com/user-attachments/assets/68e5acb1-00f4-4c89-8151-81454bfb3645)

I am just putting it out there, if sometime in the future someone smarter than me would implement such thing (if feasible) and make few more ppl happy :)

**Comment by pca006132** [comment]
Not sure what is the requirement for this. I think this is just like a typical boolean operation, we already save internal data structures with the Manifold class, e.g. collider. Maybe the simplification can get really slow when the mesh becomes very complex, even though the changes is minor?

In general, this kind of things should be optimized by the user, by splitting the scene into multiple meshes and only modify the parts that are intersecting.

**Reply by hrgdavor** [reply]
As I said, I am just putting this out there. 

You are better suited to say, if there is something manifold could provide internally to make this more performant, or like you already suggested how to optimize it from the user side.

**Comment by neilpa** [comment]
> One such case is simulating cnc tools path and how it affects a slab. It would enatial a starting geometry, that is kept in memory with associated data possible, and then produce outputs by code giving a boolean op (some shape to cut) and returning current mesh.

You might also find the [CAMotics repo](https://github.com/CauldronDevelopmentLLC/CAMotics/tree/master) interesting. In particular, [this issue](https://github.com/CauldronDevelopmentLLC/CAMotics/issues/17) has some insights into their current approach and how 5-axis sim might work.
